kube/src/config/file_config.rs

```diff
@@ -179,6 +179,65 @@ impl AuthInfo {
                 if let Some(id_token) = provider.config.get("id-token") {
                     self.token = Some(id_token.clone());
                 }
+
+                if self.token.is_none() {
+                    if let Some(cmd) = provider.config.get("cmd-path") {
+                        let params = provider.config.get("cmd-args").cloned().unwrap_or_default();
+
+                        let output = std::process::Command::new(cmd)
+                            .args(params.trim().split(" "))
+                            .output()
+                            .map_err(|e|
+                                ConfigError::AuthExec(format!("Executing {:} failed: {:?}", cmd,  e))
+                            )?;
+
+                        if !output.status.success() {
+                            return Err(ConfigError::AuthExecRun {
+                                cmd: format!{"{} {}", cmd, params},
+                                status: output.status,
+                                out: output,
+                            }.into());
+                        }
+
+                        if let Some(field) = provider.config.get("token-key") {
+                             // incredbly naive JSONPath impl
+                            let splitted = field
+                                .trim_matches(|c| c == '"' || c == '{' || c == '}')
+                                .split("." )
+                                .filter(|a| a.len() > 0);
+                            let mut current : serde_json::Value = serde_json::from_slice(&output.stdout)?;
+                            for item in splitted {
+                                if let serde_json::Value::Object(mut map) = current {
+                                    if let Some(v) = map.remove(item) {
+                                        current = v;
+                                    } else {
+                                        return Err(ConfigError::AuthExec(
+                                            format!("could not find '{:}' in result of command", item)).into());
+                                    }
+
+                                } else {
+                                    return Err(ConfigError::AuthExec(
+                                        format!("Can't lookup {:?} in result of command: not an object", item)
+                                    ).into());
+                                }
+                            }
```

Comment:
Hah, hacky. This at least makes it easy to understand, but it probably won't work well if people use a more complicated JSONPath selector?

How would you feel like using https://crates.io/crates/jsonpath_lib ? It looks like a good swap-in.
We can probably benefit from selector helpers using that anyway in the API.

Reply:
didn't want to add any dependency before sending it up, but if that is fine with you I am happy to add this.

Reply:
Please, this feels like one of those libraries that matches very well anyway :-)